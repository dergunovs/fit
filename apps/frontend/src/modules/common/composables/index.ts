import { ref, computed, onMounted } from 'vue';
import { useRouter, useRoute } from 'vue-router';

import LayoutAdmin from '@/common/components/LayoutAdmin.vue';
import LayoutDefault from '@/common/components/LayoutDefault.vue';

export function useLayout() {
  const router = useRouter();
  const route = useRoute();

  const isLoaded = ref(false);

  const layoutComponent = computed(() => (route.meta.layout === 'admin' ? LayoutAdmin : LayoutDefault));

  onMounted(async () => {
    await router.isReady();
    isLoaded.value = true;
  });

  return {
    isLoaded,
    layoutComponent,
  };
}

export function useRouteId(title: string, isQuery?: boolean) {
  const route = useRoute();

  const id = computed(() => {
    const value = isQuery ? route.query[title]?.toString() : route.params[title].toString();

    return value || '';
  });

  return { id };
}
