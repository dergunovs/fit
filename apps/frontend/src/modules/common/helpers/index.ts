import { toast } from 'mhz-ui';

export async function copyToClipboard(text: string, toastText: string) {
  await navigator.clipboard.writeText(text);

  toast.success(toastText);
}
