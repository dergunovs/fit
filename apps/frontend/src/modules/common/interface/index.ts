import { TPostAuthLoginDTO } from 'fitness-tracker-contracts';
import { FunctionalComponent } from 'vue';

export interface INavItem {
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IOnSuccess {
  create?: (id?: string) => Promise<void>;
  update?: () => Promise<void>;
  delete?: () => Promise<void>;
  login?: (user: TPostAuthLoginDTO) => Promise<void>;
  setup?: () => Promise<void>;
  register?: () => Promise<void>;
}

export interface IPromptEvent extends Event {
  prompt: () => Promise<void>;
}

export interface IGetLatestVersionDTO {
  'dist-tags': { latest: string };
}
