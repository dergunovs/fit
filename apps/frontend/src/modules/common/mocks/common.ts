import { vi } from 'vitest';

export const spyCopyToClipboard = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
