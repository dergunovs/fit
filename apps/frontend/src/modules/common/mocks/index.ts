export * from './helpers';
export * from './ui';
export * from './router';
export * from './layout';
export * from './pwa';
export * from './service';
