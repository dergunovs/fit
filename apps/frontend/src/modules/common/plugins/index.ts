import { createI18n, I18nOptions } from 'vue-i18n';

import ru from '@/common/locales/ru.json';
import en from '@/common/locales/en.json';

const messages = { en, ru } as const;

const options: I18nOptions = { legacy: false, locale: 'ru', messages };

// eslint-disable-next-line @typescript-eslint/no-type-alias
export type TMessageSchema = typeof en;

export const i18n = createI18n<false, typeof options>(options);
