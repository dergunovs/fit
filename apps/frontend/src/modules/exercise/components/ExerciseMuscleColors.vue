<template>
  <div v-if="!props.isHide" :class="$style.muscles">
    <div
      v-for="muscle in props.muscles"
      :key="muscle._id"
      :class="$style.muscle"
      :style="`background: ${muscle.color}; height: ${100 / props.muscles.length}%`"
      data-test="exercise-muscle-color"
    ></div>
  </div>
</template>

<script setup lang="ts">
import { IMuscle } from 'fitness-tracker-contracts';

interface IProps {
  muscles: IMuscle[];
  isHide?: boolean;
}

const props = defineProps<IProps>();
</script>

<style module lang="scss">
.muscles {
  position: absolute;
  top: -6px;
  left: -6px;
  width: 32px;
  height: 32px;
  rotate: 45deg;
}

.muscle {
  width: 16px;
}
</style>
