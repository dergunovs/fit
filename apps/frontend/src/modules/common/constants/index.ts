import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
