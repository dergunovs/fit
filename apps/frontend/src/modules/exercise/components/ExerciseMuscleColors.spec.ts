import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import ExerciseMuscleColors from './ExerciseMuscleColors.vue';

import { wrapperFactory } from '@/common/test';
import { MUSCLES_FIXTURE } from '@/muscle/fixtures';

const color = dataTest('exercise-muscle-color');

let wrapper: VueWrapper<InstanceType<typeof ExerciseMuscleColors>>;

beforeEach(() => {
  wrapper = wrapperFactory(ExerciseMuscleColors, { muscles: MUSCLES_FIXTURE });
});

enableAutoUnmount(afterEach);

describe('ExerciseMuscleColors', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(ExerciseMuscleColors)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows colors', async () => {
    expect(wrapper.findAll(color).length).toStrictEqual(MUSCLES_FIXTURE.length);
    expect(wrapper.find(color).attributes('style')).toStrictEqual(
      `background: ${MUSCLES_FIXTURE[0].color}; height: ${100 / MUSCLES_FIXTURE.length}%;`
    );
  });
});
