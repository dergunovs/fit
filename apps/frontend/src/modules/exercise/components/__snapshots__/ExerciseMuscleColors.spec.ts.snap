// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExerciseMuscleColors > matches snapshot 1`] = `
"<div class="_muscles_757c72">
  <div class="_muscle_757c72" style="background: #00CED1; height: 16.666666666666668%;" data-test="exercise-muscle-color"></div>
  <div class="_muscle_757c72" style="background: #0000ff; height: 16.666666666666668%;" data-test="exercise-muscle-color"></div>
  <div class="_muscle_757c72" style="background: #C41E3A; height: 16.666666666666668%;" data-test="exercise-muscle-color"></div>
  <div class="_muscle_757c72" style="background: #32CD32; height: 16.666666666666668%;" data-test="exercise-muscle-color"></div>
  <div class="_muscle_757c72" style="background: #DA70D6; height: 16.666666666666668%;" data-test="exercise-muscle-color"></div>
  <div class="_muscle_757c72" style="background: #FF8C00; height: 16.666666666666668%;" data-test="exercise-muscle-color"></div>
</div>"
`;
